pkg/deployment/note/list.go

```diff
@@ -0,0 +1,68 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package note
+
+import (
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/client/deployments"
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+
+	"github.com/elastic/ecctl/pkg/deployment"
+)
+
+// ListParams is used by List
+type ListParams struct {
+	deployment.Params
+}
+
+// Use different resource types when this is supported by the API.
+// For the time being, the notes endpoint only allows elasticsearch IDs.
+func (params *ListParams) fillDefaults() error {
+	esID, err := getElasticsearchID(deployment.GetParams{
+		API:          params.API,
+		DeploymentID: params.ID,
+	})
+	if err != nil {
+		return err
+	}
+
+	params.ID = esID
+	return err
```

Comment:
It's preferable to return the explicit nil since above `err != nil ` is already checked for. Otherwise always returning `err` without checking if `err != nil` is also less code

```suggestion
	return nil
```

Reply:
😄 I have no idea why I did that